pennylane/math/single_dispatch.py

```diff
@@ -33,6 +33,11 @@ def _i(name):
     return import_module(name)
 
 
+# ------------------------------- Builtins -------------------------------- #
+
+ar.register_function("builtins", "ndim", lambda x: np.ndim(np.array(x)))
+ar.register_function("builtins", "shape", lambda x: np.array(x).shape)

```

Comment:
We could use `np.array(x).ndim` for `ndim`, similar to `shape`. Is there a reason to prefer one over the other (I'd guess attribute lookup should be faster?)

Reply:
It works for now.